src/main/java/com/lorepo/icplayer/client/page/PageController.java

```diff
@@ -608,10 +654,51 @@ public HashMap<String, Widget> getWidgets() {
 		return null;
 	}
 
-	public void outstretchHeight(int y, int height, boolean dontMoveModules) {
-		this.outstretchHeightWithoutAddingToModifications(y, height, false, dontMoveModules);
-		this.currentPage.heightModifications.addOutstretchHeight(y, height, dontMoveModules);
-		this.playerController.fireOutstretchHeightEvent();
+	public void outstretchHeight(int y, int height, boolean dontMoveModules, String layoutName) {
+		String actualLayoutName = this.getActualLayoutName();
+		this.updateOutstretchDictionary(y, height, dontMoveModules, layoutName);
+
+		if (actualLayoutName.equals(layoutName) || layoutName.isEmpty()) {
+			this.outstretchHeightWithoutAddingToModifications(y, height, false, dontMoveModules);
+			this.currentPage.heightModifications.addOutstretchHeight(y, height, dontMoveModules);
+			this.playerController.fireOutstretchHeightEvent();
+			this.previousLayoutName = actualLayoutName;
+		}
+	}
+
+	private boolean hasOppositeValue(int height, String layoutName) {
+		String layoutKey = this.getLayoutKey(layoutName);
+		
+		if (!this.contentModel.isOutstretchPageDictionaryEmpty() && this.contentModel.hasOutstretchPage(layoutKey)) {
+			OutstretchPageHeight parameter = contentModel.getOutstretchPage(layoutKey);
+			return parameter.getValueOfStretching() == -height;
+		}
+
+		return false;
+	}
+
+	private String getActualLayoutName() {
+		String layoutID = this.contentModel.getActualSemiResponsiveLayoutID();
+		return this.contentModel.getLayoutNameByID(layoutID);
+	}
+
+	private String getLayoutKey(String layoutName) {
+		String layoutKey = layoutName
+			.concat("_")
+			.concat(this.currentPage.getName().replaceAll(" ", "_"));
+		
+			return layoutKey;
```

Comment:
Niepotrzebny dodatkowy indent
```suggestion
		return layoutKey;
```